flexbox/src/main/java/com/google/android/flexbox/FlexboxLayout.java

```diff
@@ -1367,14 +1479,31 @@ private void setMeasuredDimensionForFlex(@FlexDirection int flexDirection, int w
      * @see #getFlexWrap()
      * @see #setFlexWrap(int)
      */
-    private boolean isWrapRequired(int flexWrap, int mode, int maxSize,
-            int currentLength, int childLength, LayoutParams lp) {
-        if (flexWrap == FLEX_WRAP_NOWRAP) {
+    private boolean isWrapRequired(int mode, int maxSize, int currentLength, int childLength,
+            LayoutParams lp, int childAbsoluteIndex, int childRelativeIndexInFlexLine) {
+        if (mFlexWrap == FLEX_WRAP_NOWRAP) {
             return false;
         }
         if (lp.wrapBefore) {
             return true;
         }
+        if (isMainAxisDirectionHorizontal(mFlexDirection)) {
+            if (hasDividerBeforeChildAtAlongMainAxis(childAbsoluteIndex,
+                    childRelativeIndexInFlexLine)) {
+                childLength += mDividerVerticalWidth;
+            }
+            if ((mShowDividerVertical & SHOW_DIVIDER_END) > 0) {
+                childLength += mDividerVerticalWidth;
+            }
+        } else {
+            if (hasDividerBeforeChildAtAlongMainAxis(childAbsoluteIndex,
+                    childRelativeIndexInFlexLine)) {
+                childLength += mDividerHorizontalHeight;
+            }
+            if ((mShowDividerHorizontal & SHOW_DIVIDER_END) > 0) {
+                childLength += mDividerHorizontalHeight;
+            }
+        }
         return (mode == MeasureSpec.EXACTLY || mode == MeasureSpec.AT_MOST) &&
```

Comment:
done. Actually when the `If  (mode == MeasureSpec.EXACTLY || mode == MeasureSpec.AT_MOST)` condition is <b>false</b>, the calc above is not needed.

Anyway changed the code to avoid the calc when `mode != MeasureSpec.UNSPECIFIED`